From dab1854e5ec0a91fd28c9d84f06dcdd0af39518b Mon Sep 17 00:00:00 2001
From: OPNA2608 <opna2608@protonmail.com>
Date: Sat, 7 Feb 2026 11:18:02 +0100
Subject: [PATCH] Properly include lomiri-api includedirs

In lomiri-api 0.3.0, liblomiri-api includedir was moved to a subdir of <...>/include.

The includes here previously only worked because <...>/include is often put into the compiler's
list of default include dirs. Explicitly include the correct dirs instead.
---
 src/com/lomiri/content/service/CMakeLists.txt | 1 +
 tests/acceptance-tests/CMakeLists.txt         | 4 ++--
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/com/lomiri/content/service/CMakeLists.txt b/src/com/lomiri/content/service/CMakeLists.txt
index f85c5e0..ba0d180 100644
--- a/src/com/lomiri/content/service/CMakeLists.txt
+++ b/src/com/lomiri/content/service/CMakeLists.txt
@@ -31,6 +31,7 @@ include_directories(
   ${CMAKE_SOURCE_DIR}/src/com/lomiri/content
   ${GIO_INCLUDE_DIRS}
   ${GSETTINGS_INCLUDE_DIRS}
+  ${LOMIRI_API_INCLUDE_DIRS}
   ${LOMIRI_DOWNLOAD_MANAGER_INCLUDE_DIRS}
   ${LOMIRI_LAUNCH_INCLUDE_DIRS}
   ${LIBERTINE_INCLUDE_DIRS}
diff --git a/tests/acceptance-tests/CMakeLists.txt b/tests/acceptance-tests/CMakeLists.txt
index 4fbd447..33cad67 100644
--- a/tests/acceptance-tests/CMakeLists.txt
+++ b/tests/acceptance-tests/CMakeLists.txt
@@ -51,7 +51,7 @@ foreach(test ${TESTS})
     ${${test}_SOURCES}
   )
   find_package(Qt${QT_VERSION_MAJOR} REQUIRED COMPONENTS Test)
-  target_include_directories(${test} PRIVATE ${LOMIRI_DOWNLOAD_MANAGER_INCLUDE_DIRS})
+  target_include_directories(${test} PRIVATE ${LOMIRI_API_INCLUDE_DIRS} ${LOMIRI_DOWNLOAD_MANAGER_INCLUDE_DIRS})
   target_link_libraries(${test} Qt::Core Qt::Gui Qt::DBus Qt::Test)
 
   target_link_libraries(${test}
@@ -77,7 +77,7 @@ add_executable(
   source_all.json
 )
 
-target_include_directories(test_registry_updater PRIVATE ${GSETTINGS_INCLUDE_DIRS})
+target_include_directories(test_registry_updater PRIVATE ${LOMIRI_API_INCLUDE_DIRS} ${GSETTINGS_INCLUDE_DIRS})
 target_link_libraries(test_registry_updater Qt::Core Qt::Gui Qt::DBus Qt::Test)
 target_link_libraries(test_registry_updater ${TEST_LIBS} ${GSETTINGS_LDFLAGS})
 add_test(NAME test_registry_updater COMMAND dbus-test-runner --task ${CMAKE_CURRENT_BINARY_DIR}/test_registry_updater)
-- 
2.53.0

